File: app/src/main/java/com/alexkn/syntact/data/util/Converters.java
package com.alexkn.syntact.data.util;

import androidx.room.TypeConverter;

import com.alexkn.syntact.restservice.TemplateType;

import java.time.Instant;
import java.util.Locale;

public class Converters {

    @TypeConverter
    public static Instant toInstant(Long value) {

        return value == null ? null : Instant.ofEpochMilli(value);
    }

    @TypeConverter
    public static Long toLong(Instant date) {

        return date == null ? null : date.toEpochMilli();
    }

    @TypeConverter
    public static Locale toLocale(String value) {

        return value == null ? null : new Locale(value);
    }

    @TypeConverter
    public static String toString(Locale locale) {

        return locale == null ? null : locale.getLanguage();
    }

    @TypeConverter
    public static TemplateType toTemplateType(String value) {

        return value == null ? null : TemplateType.valueOf(value);
    }

    @TypeConverter
    public static String toString(TemplateType templateType) {

        return templateType == null ? null : templateType.name();
    }
}

File: app/src/main/java/com/alexkn/syntact/data/util/Converters.kt
package com.alexkn.syntact.data.util

import androidx.room.TypeConverter

import com.alexkn.syntact.data.common.TemplateType

import java.time.Instant
import java.util.Locale

class Converters {

    @TypeConverter
    fun toInstant(value: Long?): Instant? = value?.let { Instant.ofEpochMilli(value) }

    @TypeConverter
    fun toLong(date: Instant?) = date?.toEpochMilli()

    @TypeConverter
    fun toLocale(value: String?) = value?.let { Locale(value) }

    @TypeConverter
    fun toString(locale: Locale?) = locale?.language

    @TypeConverter
    fun toTemplateType(value: String?) = value?.let(TemplateType::valueOf)

    @TypeConverter
    fun toString(templateType: TemplateType?) = templateType?.name
}



